### ISSUE TITLE
Batch insert, I have used the transaction, but the speed is not very satisfactory

### ISSUE THREAD

**Issue by FingersKey** [neo4jrb/neo4j#1332]
This is the way I used it.

```
Neo4j::Transaction.run do
  items.each do |item|
    # Build relationships .........
    item.save(validate: false)
  end
end
```

I insert data > = 10000, which usually takes about 10 minutes

Is there any faster way!!!
Has anyone else run into this / have other suggestions?

 

### Runtime information:

Neo4j database version:
`neo4j` gem version: '>=7.0.1'
`neo4j-core` gem version:

**Comment by cheerfulstoic** [comment]
I'm assuming that `items` are `ActiveRecord` objects.  At the moment there isn't a faster way to save `ActiveRecord` objects in a batch like that.  I do have a few suggestions, though:

 * If you use version `8.0.0` of the gem you can configure the bolt protocol, which *may* be faster in this case
 * Starting with version 7.0.0 of the `neo4j-core` gem (which is included when you use the 8.0.0 of the `neo4j` gem) introduces a way to make batch queries over HTTP.  Just using a transaction isn't sufficient for HTTP because each request is still an individual HTTP request (one to start the transaction, one to make the first query, etc...).  In the new `Neo4j::Core::CypherSession` API from `neo4j-core` (see the [repo README](https://github.com/neo4jrb/neo4j-core)) you can use the `queries` method to make multiple queries in one request.  The drawback there is that you need to write your own Cypher
 * You might also peek at the [neo4apis](https://github.com/neo4jrb/neo4apis) gem which is a gem which allows you to write a DSL for importing objects efficiently in batches (you might look at [neo4apis-twitter](https://github.com/neo4jrb/neo4apis-twitter) or [neo4apis-github](https://github.com/neo4jrb/neo4apis-github) for examples on how this is done

If you upgrade to 8.0 of `neo4j` / 7.0 of `neo4j-core` you should probably check out the upgrade guide:

http://neo4jrb.readthedocs.io/en/8.0.x/UpgradeGuide.html

Lastly, if you have any questions drop by our [Gitter channel](https://gitter.im/neo4jrb/neo4j) or ask a question on [StackOverflow](http://stackoverflow.com/questions/ask?tags=neo4j.rb+neo4j+ruby).  Since this isn't a bug or a feature request I'm going to close it

**Comment by cheerfulstoic** [comment]
This is a pretty common question, so I created a page in our docs for it: http://neo4jrb.readthedocs.io/en/8.0.x/Miscellany.html#mass-batch-importing
